nilearn/plotting/displays/_figures.py

```diff
@@ -103,3 +113,141 @@ def savefig(self, output_file=None):
         self._check_output_file(output_file=output_file)
         if self.figure is not None:
             self.figure.write_image(self.output_file)
+
+    def add_contours(self, roi_map, levels=None, labels=None, lines=None):
+        """
+        Draw boundaries around roi.
+
+        Parameters
+        ----------
+        roi_map : str or :class:`numpy.ndarray` or list of
+            :class:`numpy.ndarray` ROI map to be displayed on the surface
+            mesh, can be a file (valid formats are .gii, .mgz, .nii,
+            .nii.gz, or Freesurfer specific files such as .annot or .label),
+            or a Numpy array with a value for each vertex of the surf_mesh.
+            The value at each vertex one inside the ROI and zero inside ROI,
+            or an integer giving the label number for atlases.
+
+        levels : list of integers, or None, optional
+            A list of indices of the regions that are to be outlined.
+            Every index needs to correspond to one index in roi_map.
+            If None, all regions in roi_map are used.
+
+        labels : list of strings or None, or None, optional
+            A list of labels for the individual regions of interest. Provide
+            None as list entry to skip showing the label of that region. If
+            None, no labels are used.
+
+        lines : list of dict giving the properties of the contours, or None,
+            optional. For valid keys, see
+            :attr:`plotly.graph_objects.Scatter3d.line`. If length 1, the
+            properties defined in that element will be used to draw all
+            requested contours.
+        """
+        import plotly.graph_objects as go
```

Comment:
is this import here for a specific reason (avoid circular import for example) otherwise you can definitely put it at the top of the module

Reply:
Importing plotly at the top caused this workflow to fail: https://github.com/nilearn/nilearn/blob/1759fa2f719e8eb56627698484f8665cc5336824/.github/workflows/testing.yml#L12-L21. I think that workflow brings in just these packages: https://github.com/nilearn/nilearn/blob/1759fa2f719e8eb56627698484f8665cc5336824/pyproject.toml#L77-L84. Is there a better way to import?

Reply:
oops my bad.

In this case you want to use a try block for more gracefull failures in case the import fails if plotly is not around

https://github.com/nilearn/nilearn/blob/1759fa2f719e8eb56627698484f8665cc5336824/nilearn/plotting/surf_plotting.py#L252C1-L258C31

Reply:
> n this case you want to use a try block

Actually this may be an overkill given that this is part of the a class where the constructor already has such a check.

Reply:
I suppose there could be a try/except at the top of the file, replacing all instances of `import plotly.graph_objects` in the methods? If the import error should still be raised within the constructor, then perhaps the approach used in the tests could be replicated https://github.com/nilearn/nilearn/blob/1759fa2f719e8eb56627698484f8665cc5336824/nilearn/plotting/tests/test_surf_plotting.py#L30-L35

e.g., 
```python
if not PLOTLY_INSTALLED:
    raise ImportError("Plotly is required to use `PlotlySurfaceFigure`.")

```

Reply:
yeah I had seen this one

Wonder if some of this should not be extracted into a separate utility function `is_plotly_installed()`. @ymzayek WDYT?

this should probably be done in a different PR though, so don't overthing it for this PR

Reply:
Sure it doesn't hurt but yea I would leave it for another PR. Either the current way or what's suggested [here](https://github.com/nilearn/nilearn/pull/3949#discussion_r1317441527) is fine for this PR.

Reply:
[e3fc2bb](https://github.com/nilearn/nilearn/pull/3949/commits/e3fc2bbc29538febc31e7f9fa71a1b762cae444f) does it